packages/dashboard-v2/src/components/forms/AccountSettingsForm.js

```diff
@@ -0,0 +1,106 @@
+import * as Yup from "yup";
+import PropTypes from "prop-types";
+import { Formik, Form } from "formik";
+
+import { Button } from "../Button";
+import { TextField } from "../Form/TextField";
+import accountsService from "../../services/accountsService";
+
+const isPopulated = (value) => value?.length > 0;
+
+const emailUpdateSchema = Yup.object().shape({
+  email: Yup.string().email("Please provide a valid email address"),
+  confirmEmail: Yup.string()
+    .oneOf([Yup.ref("email"), null], "Emails must match")
+    .when("email", {
+      is: isPopulated,
+      then: (schema) => schema.required("Please confirm new email address"),
+    }),
+  password: Yup.string().min(6, "Password has to be at least 6 characters long"),
+  confirmPassword: Yup.string()
+    .oneOf([Yup.ref("password"), null], "Passwords must match")
+    .when("password", {
+      is: isPopulated,
+      then: (schema) => schema.required("Please confirm new password"),
+    }),
+});
+
+export const AccountSettingsForm = ({ user, onSuccess, onFailure }) => {
+  return (
+    <Formik
+      initialValues={{
+        email: "",
+        confirmEmail: "",
+        password: "",
+        confirmPassword: "",
+      }}
+      validationSchema={emailUpdateSchema}
+      onSubmit={async ({ email, password }, { resetForm }) => {
+        try {
+          await accountsService.put("user", {
+            json: { email, password },
+          });
```

Comment:
should we only send email / password if it's supplied? it seems like it would send empty string instead of email if we don't supply it ?

Reply:
Yes, but blank values seem to be ignored by the API right now. That said, emails will become optional, so here's a bulletproofing patch: 52314130